src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneIndexManager.cs

```diff
@@ -218,7 +218,7 @@ private Document CreateLuceneDocument(DocumentIndex documentIndex)
                         break;
 
                     case DocumentIndex.Types.Integer:
-                        if (entry.Value != null)
+                        if (entry.Value != null && Int32.TryParse(entry.Value?.ToString(), out var value))
```

Comment:
If the `value` checked for nullability in the first condition, I don't think that `entry.Value?.ToString()` check nullability again is required

Reply:
You are right.